app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/hplus/HPlusSupport.java

```diff
@@ -636,6 +636,79 @@ public void onTestNewFunction() {
     }
 
 
+    private void showIncomingCall(String name, String number){
+        LOG.debug("Show Incoming Call");
+
+        try {
+            TransactionBuilder builder = performInitialized("incomingCallIcon");
+
+            //Enable call notifications
+            builder.write(ctrlCharacteristic, new byte[] {HPlusConstants.COMMAND_SET_INCOMING_CALL, 1 });
+
+            //Show Call Icon
+            builder.write(ctrlCharacteristic, HPlusConstants.COMMAND_ACTION_INCOMING_CALL);
+
+            //builder = performInitialized("incomingCallText");
+            builder.queue(getQueue());
+
+            try {
+                Thread.sleep(200);
```

Comment:
I will use the WaitAction. Apparently the device will loose messages if they are sent without delay. There is no feedback to trigger a new message.